### PR TITLE
Add `parent` project

### PR DESCRIPTION
**Pull request by melix**
This commit introduces the `parent` project as part of the BOM project. This project used to live in `core` but has been moved to this project instead, since it participates in releases independently of `core`.

As part of the process, the way the parent POM file has been generated moves from a Groovy DSL based POM generator to a hybrid approach:

- a template pom.xml file is found in `src/pom-template.xml`
- properties are extracted into `libs.versions.toml` so that they can be automatically upgraded by Renovatebot
- some properties are inlined from included BOMs (grpc, picocli)

A "maven-pom" plugin has been created so that if we want to generate a new parent POM, say for libraries, it should be fairly straightforward.

### CODE REVIEW

**Comment by melix** [comment]
One this is merged, the `parent` project in `core` will have to be removed. I will prepare a PR.

File: parent/src/pom-template.xml [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    </scm>
+    <developers>
+        <developer>
+            <id>graemerocher</id>
+            <name>Graeme Rocher</name>
+        </developer>
+    </developers>
+    <properties>
+        <jdk.version>1.8</jdk.version>
+        <release.version>8</release.version>
```
Comment (graemerocher) [inline parent/src/pom-template.xml:37]: this needs updating to 17

File: parent/src/pom-template.xml [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    <profiles>
+        <profile>
+            <id>jdk-9-or-later</id>
+            <activation>
+                <jdk>[9,)</jdk>
+            </activation>
+            <properties>
+                <maven.compiler.release>${release.version}</maven.compiler.release>
+            </properties>
+        </profile>
```
Comment (graemerocher) [inline parent/src/pom-template.xml:55]: This profile shouldn't be needed anymore, how is it used?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: parent/src/pom-template.xml (hunk 1 above)
Comment (melix) [reply parent/src/pom-template.xml:37]: Good catch

File: parent/src/pom-template.xml (hunk 2 above)
Comment (melix) [reply parent/src/pom-template.xml]: I have pushed a change, but I think it probably doesn't make sense to have both `<maven.compiler.release>${release.version}</maven.compiler.release>` and the source/target compatibility set. If the previous profile was using `release`, then we should probably only use `release` too.

**Comment by graemerocher** [comment]
@alvarosanchez can you check this parent is appropriate?

**Comment by alvarosanchez** [comment]
> @alvarosanchez can you check this parent is appropriate?

If other than the Java version, it produces the same parent as we have now, then yes, it's correct.

Once I get to work on Maven plugin for Micronaut 4, it may require changes, but I'll handle them from here.

File: parent/src/pom-template.xml [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                                    </transformer>
+                                    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                                </transformers>
+                                <filters>
+                                    <filter>
+                                        <artifact>*.*</artifact>
+                                        <excludes>
+                                            <exclude>META-INF/*.SF</exclude>
+                                            <exclude>META-INF/*.DSA</exclude>
+                                            <exclude>META-INF/*.RSA</exclude>
```
Comment (alvarosanchez) [inline parent/src/pom-template.xml:228]:
```suggestion
                                            <exclude>META-INF/*.RSA</exclude>
                                            <exclude>module-info.class</exclude>
```

**Review by alvarosanchez — APPROVED** [review]
Added a suggestion to account for https://github.com/micronaut-projects/micronaut-core/issues/8402

**Pull request merged**